### PR TITLE
remove more usages of any

### PR DESCRIPTION
**Pull request by bafolts**
Removes more usages of `any` from the code.

### CODE REVIEW

File: src/client/directives/TrimWhitespace.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 
-export function trimEmptyTextNodes(el: any) {
-  for (const node of el.childNodes) {
-    if (node.nodeType === Node.TEXT_NODE && node.data.trim() === '') {
+export function trimEmptyTextNodes(el: Node) {
+  for (let i = 0; i < el.childNodes.length; i++) {
```
Comment (bafolts) [inline src/client/directives/TrimWhitespace.ts:3]: Can't use `for...of` on HTML collection.

Comment (kberg) [reply src/client/directives/TrimWhitespace.ts:3]: > Can't use `for...of` on HTML collection.
And yet it somewhow worked?

File: src/client/components/PreferencesDialog.vue [hunk 2]
```diff
@@ -59,7 +59,7 @@ export default Vue.extend({
       }
       return PreferencesManager.preferencesValues;
     },
-    updatePreferences(_evt: any): void {
```
Comment (kberg) [inline src/client/components/PreferencesDialog.vue:62]: any `any` is one `any` too many.

**Review by kberg — APPROVED** [review]

File: src/client/directives/TrimWhitespace.ts (hunk 1 above)
Comment (bafolts) [reply src/client/directives/TrimWhitespace.ts:3]: The generated javascript here may have gotten lucky that it was using the `.length` property which exists on an HTML collection. What that `for...of` would get wrong though is that an HTML Collection is a _live list_. As items are removed from the DOM it is updated in real time. As we were calling to `remove` we were most likely skipping items.

**Pull request merged**